rosetta-ide/src/main/java/com/regnosys/rosetta/ide/inlayhints/IInlayHintsResolver.java

```diff
@@ -1,3 +1,19 @@
+/*
+ * Copyright 2024 REGnosys
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
Question about this header. Do we need to copy this manually every time we create a class in Eclipse? Or is there an automation for this? Is this validated somehow?

Reply:
There are some tools to help you aligning the file header, at different levels:
- Maven - enforces headers at build time
- Pre-commit hooks - enforces headers at commit time
- IDE integration - one button to re-generate all headers

Up to you folks to decide what best suites your dev cycles.

Re. the importance of these headers, there are different opinions throughout the OSS community.

If you want to preserve attribution, and avoid brutal copy/pasting (or AI explotation with no attribution), granted that the Apache License allows for this usage (!!!), then I'd suggest using file headers across the entire codebase.